src/rocksdb/tests/basic.rs

```diff
@@ -123,6 +128,127 @@ async fn basic() -> Result<(), anyhow::Error> {
+
+        for op in operands {
+            val += op;
+        }
+        Some(val)
+    }
+
+    let static_options = InstanceOptions::new(
+        rocksdb::Env::new()?,
+        2,
+        Some(("test".to_string(), merge)),
+        bincode::DefaultOptions::new(),
+    );
+
+    let mut instance = RocksDBInstance::<String, u32>::new(
+        t.path(),
+        static_options,
+        RocksDBConfig::new(Default::default(), None),
+        shared_metrics_for_tests()?,
+        instance_metrics_for_tests()?,
+    )
+    .await?;
+
+    let mut rolling_sum = 50;
+    let key = "a".to_string();
+    // Set the key to an initial value using Put
+    instance
+        .multi_update(vec![(key.clone(), KeyUpdate::Put(rolling_sum))])
+        .await?;
+
+    // Send a bunch of merge operations to the key, each merge transaction
+    // should add 25 million to the key value, and every 10 transactions
+    // we use a Put to set the key to a specific value.
+    for i in 1..100 {
+        let merges = vec![5u32, 5_000_000];
+        rolling_sum += merges.iter().sum::<u32>();
+
+        instance
+            .multi_update(
+                merges
+                    .into_iter()
+                    .map(|v| (key.clone(), KeyUpdate::Merge(v))),
+            )
+            .await?;
+
+        if i % 10 == 0 {
+            rolling_sum += 3;
+            instance
+                .multi_update(vec![(key.clone(), KeyUpdate::Put(rolling_sum))])
+                .await?;
+        }
+    }
+
+    let mut ret = vec![Default::default(); 1];
+    instance
+        .multi_get(vec![key.clone()], ret.iter_mut(), |value| value)
+        .await?;
+
+    // Validate the Get operation returns the correct sum
+    assert_eq!(
+        ret.into_iter()
+            .map(|v| v.map(|v| v.value))
+            .collect::<Vec<_>>(),
+        vec![Some(rolling_sum)]
+    );
+
+    // Send more merge/put operations, this time compacting between each transaction
+    for i in 1..100 {
+        let merges = vec![5u32, 5_000_000];
+        rolling_sum += merges.iter().sum::<u32>();
+
+        instance
+            .multi_update(
+                merges
+                    .into_iter()
+                    .map(|v| (key.clone(), KeyUpdate::Merge(v))),
+            )
+            .await?;
+
+        instance.manual_compaction().await?;
+
+        if i % 10 == 0 {
+            rolling_sum += 3;
+            instance
+                .multi_update(vec![(key.clone(), KeyUpdate::Put(rolling_sum))])
+                .await?;
+        }
+    }
+
+    let mut ret = vec![Default::default(); 1];
+    instance
+        .multi_get(vec![key.clone()], ret.iter_mut(), |value| value)
+        .await?;
+
+    // Validate the Get operation returns the correct sum
+    assert_eq!(
+        ret.into_iter()
+            .map(|v| v.map(|v| v.value))
+            .collect::<Vec<_>>(),
+        vec![Some(rolling_sum)]
+    );
+
+    instance.close().await?;
+
+    Ok(())
+}
```

Comment:
can we also have a test that just merges? I don't see an actual assert that we `multi_get` a merged value? Just the puts that overwrite the value?

Reply:
Ah yes good catch! I think I lost this after refactoring the tests to validate that `Put` resets the keys too. I'll rework this to ensure we validate the pure-merge workload too

Reply:
thanks!

And a merge-then-put-overriding workload, which is the upsert case!